modules/oe_theme_helper/src/ExternalLinks.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Url;
+
+/**
+ * Verifies if a URL is considered external or internal.
+ */
+class ExternalLinks implements ExternalLinksInterface {
+
+  /**
+   * The configuration factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Constructs an ExternalLinks object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The configuration factory.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isExternalLink($url): bool {
```

Comment:
We should cover this with a dedicated test, where we throw at it all sorts of different values, of different types, and check the output.